app/http.php

```diff
@@ -230,18 +230,44 @@ function logError(Throwable $error, string $action, ?Logger $logger, Utopia\Rout
     });
 
 App::wildcard()
-    ->inject('balancerGroup')
+    ->inject('balancer')
+    ->inject('state')
     ->inject('request')
     ->inject('response')
-    ->action(function (Group $balancer, Request $request, Response $response) {
+    ->action(function (Group $balancer, Table $state, Request $request, Response $response) {
         $option = $balancer->run();
 
         if ($option === null) {
             throw new Exception('No online executor found', 404);
         }
 
+        /**
+         * @var string $hostname
+         */
         $hostname = $option->getState('hostname') ?? '';
 
+        // Optimistic update. Mark runtime up instantly to prevent race conditions
+        // Next health check with confirm it started well, and update usage stats
+        $runtimeId = $request->getHeader('x-opr-runtime-id', '');
+        if (!empty($runtimeId)) {
+            /**
+             * @var array<string,mixed> $stateItem
+             */
+            $stateItem = \json_decode($state->get($hostname)['state'] ?? '{}', true);
+
+            if ($stateItem['runtimes'] === null) {
```

Comment:
```suggestion
            if (!isset($stateItem['runtimes'])) {
```
You can bring the phpstan level back to max and use `isset` instead

Reply:
Used isset, PHPstan still complains. Level 8 is fine for now.